bundles/org.openhab.core.config.discovery/src/main/java/org/openhab/core/config/discovery/AbstractDiscoveryService.java

```diff
@@ -133,6 +162,21 @@ public Set<ThingTypeUID> getSupportedThingTypes() {
         return supportedThingTypes;
     }
 
+    @Override
+    public boolean isScanInputSupported() {
+        return scanInputLabel != null && scanInputDescription != null;
```

Comment:
@lolodomo 
I think the goal of this implementation was to automatically return true if label and description are provided.
This however does not work at the moment, you rather need this:

```java
        return getScanInputLabel() != null && getScanInputDescription() != null;
```

Reply:
What is the difference?

Reply:
When using the methods instead of the classes private fields, `isScanInputSupported` will return true if both the label and the description methods are overwritten in a child class. With the current implementation, this is not the case, thus requiring a child class that already provides label and description trough the methods to also overwrite the `isScanInputSupported` method.
I will create a PR to fix that.

Reply:
#4393 is merged

Reply:
For your information, the idea is absolutely not to override getScanInputLabel and getScanInputDescription in a binding, but only to provide label and description in the constructor of the binding discovery service.

Reply:
Makes sense ... sorry for the chaos.

Reply:
By the way, your change is not a problem ;)